WcaOnRails/app/models/registration.rb

```diff
@@ -25,13 +25,23 @@ class Registration < ApplicationRecord
 
   validates_numericality_of :guests, greater_than_or_equal_to: 0
 
+  validates_numericality_of :guests, less_than_or_equal_to: :guest_limit, if: :check_guest_limit?
+
   validate :registration_cannot_be_deleted_and_accepted_simultaneously
   private def registration_cannot_be_deleted_and_accepted_simultaneously
     if deleted? && accepted?
       errors.add(:registration_competition_events, I18n.t('registrations.errors.cannot_be_deleted_and_accepted'))
     end
   end
 
+  def guest_limit
+    competition.guests_per_registration_limit
+  end
+
+  def check_guest_limit?
+    !!competition&.guests_per_registration_limit_enabled?
```

Comment:
Two observations:
1. `!!` double negation is a bastard child of JS. Please use Ruby idioms like `my_value.present?` instead.
2. You assume `competition` might be nil by virtue of the nil-safe `&.` operator. In the `guest_limit` method above, you do not make that same assumption. Why?

Reply:
1. Will do.
2. This was for a test case where the competition DNE. `guest_limit` is accessed after the `check_guest_limit?` conditional passes. The more I think about it, the test case is nonsensical (why would we have a registration for a competition that DNE?). Will revise.

Reply:
Re 2: I removed my (nonsensical) tests, but a check on `competition` is still needed to satisfy these existing tests:

https://github.com/thewca/worldcubeassociation.org/blob/947b0a65e1b576472c10d2c5bf38a73a281a01bb/WcaOnRails/spec/models/registration_spec.rb#L12-L22

Will use `competition.present?`